Pilot/proxyTools.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+import re
+from base64 import b16decode
+from subprocess import check_output, Popen, PIPE
+
+VOMS_FQANS_OID = b"1.3.6.1.4.1.8005.100.100.4"
+VOMS_EXTENSION_OID = b"1.3.6.1.4.1.8005.100.100.5"
+RE_OPENSSL_ANS1_FORMAT = re.compile(rb"^\s*\d+:d=(\d+)\s+hl=")
+
+
+def parseASN1(data):
+    cmd = ["openssl", "asn1parse", "-inform", "der"]
+    proc = Popen(cmd, stdin=PIPE, stdout=PIPE)
+    out, err = proc.communicate(data)
+    return out.split(b"\n")
+
+
+def findExtension(oid, lines):
+    for i, line in enumerate(lines):
+        if oid in line:
+            return i
+
+
+def getVO(proxy_data):
```

Comment:
This function should have unit tests assoicated with it (preferably with fake proxies of a few different levels, `voms-proxy-fake` should be able to help generate them).

Reply:
I have written a simple test to see if a `voms-proxy-fake` creates a proxy and tested it a an ubuntu container. 
I would also need a set of fake certificates with are in Dirac here: `/DIRAC/src/DIRAC/Core/Security/test/certs`
A change which I had to make was to install voms with `conda install -c conda-forge voms`, which would go right after the line:
[https://github.com/DIRACGrid/Pilot/blob/3a837eccbf9f494d16e504699db1e32c3dc683d5/.github/workflows/basic.yml#L31
](https://github.com/DIRACGrid/Pilot/blob/3a837eccbf9f494d16e504699db1e32c3dc683d5/.github/workflows/basic.yml#L31)

However there seem to be an issue with running `pytest.` The line:
[https://github.com/DIRACGrid/Pilot/blob/3a837eccbf9f494d16e504699db1e32c3dc683d5/.github/workflows/basic.yml#L41
](https://github.com/DIRACGrid/Pilot/blob/3a837eccbf9f494d16e504699db1e32c3dc683d5/.github/workflows/basic.yml#L41)

prevents `pytest` from being run, doesn't it ?